advanced/multipage-testing-example/automation.js

```diff
@@ -27,32 +40,47 @@ const NETWORK_CONDITIONS = {
   }
 };
 
+
 (async () => {
   console.log('Navigation script');
 
-  // Launch browser
-  const browser = await puppeteer.launch();
+  // Launch browser, use headful mode to allow Cumulative Layout Shift (CLS) measurment
+  const browser = await puppeteer.launch({headless: false, defaultViewport: null});
 
-  for (let id = 0; id < SAMPLE_SIZE; id++) {
-    console.log(`Running ${id} of ${SAMPLE_SIZE}`);
+  for (; i < SAMPLE_SIZE; i++) {
+    console.log(`Running ${i} of ${SAMPLE_SIZE}`);
 
     // Open new page
     const page = await browser.newPage();
 
     // Emulate network conditions and disable cache to simulate new users
     await page.setCacheEnabled(false);
-    await page.emulateNetworkConditions(NETWORK_CONDITIONS.Good3G);
+
+    // Evaluate Split flag to determine what the emulated network conditions should be for this user
+    const networkSpeed  = splitClient.getTreatment(i, process.env.FEATURE_FLAG_NETWORK_SPEED);
+
+    await page.emulateNetworkConditions(
+      (networkSpeed in NETWORK_CONDITIONS)
+      ? NETWORK_CONDITIONS[networkSpeed]
+      : NETWORK_CONDITIONS.Good3G
+    );
 
     // Navigate to URL
-    await page.goto(`http://localhost:3000/?id=${id}`);
+    await page.goto (`http://localhost:3000/?id=${i}` , {waitUntil: "networkidle0", timeout: 0}); // Disabled timeout to avoid exception being thrown. If, however, the page gets 'stuck', click the refresh button.
```

Comment:
```suggestion
    await page.goto(`http://localhost:3000/?id=${i}` , {waitUntil: "networkidle0", timeout: 0}); // Disabled timeout to avoid exception being thrown. If, however, the page gets 'stuck', click the refresh button.
```
as per Emi's suggestion